cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/SetPreferenceModal.js

```diff
@@ -304,53 +322,57 @@ export default class SetPreferenceModal extends Component {
   }
 
   renderInheritedPreferences() {
-    const titleLabel = T.translate('features.FastAction.setPreferencesInheritedPrefsLabel');
-    const keyLabel = T.translate('features.FastAction.setPreferencesColumnLabel.key');
-    const valueLabel = T.translate('features.FastAction.setPreferencesColumnLabel.value');
-    const originLabel = T.translate('features.FastAction.setPreferencesColumnLabel.origin');
-    let numInheritedPreferences = this.state.inheritedPreferences.length;
-    return (
-      <div>
-        <div className='inherited-preferences-label'>
-          <h4>{titleLabel} ({numInheritedPreferences})</h4>
-        </div>
-        <div className='inherited-preferences-list'>
-        {
-           numInheritedPreferences ?
-            <div>
-              <table>
-                <thead>
-                  <tr>
-                    {this.renderInheritedPreferencesColumnHeader(keyLabel)}
-                    {this.renderInheritedPreferencesColumnHeader(valueLabel)}
-                    <th>{originLabel}</th>
-                  </tr>
-                </thead>
-                <tbody>
-                  {
-                    this.state
-                      .inheritedPreferences
-                      .map((inheritedPreference, index) => {
-                        return (
-                          <tr className="inherited-preference" key={index}>
-                            <td>{inheritedPreference.key}</td>
-                            <td>{inheritedPreference.value}</td>
-                            <td>N/A</td>
-                          </tr>
-                        );
-                      })
-                  }
-                </tbody>
-              </table>
-            </div>
-          :
-            <div className="text-xs-center">
-              No Inherited Preferences
-            </div>
-        }
+    if (this.getInheritedPreferencesApi) {
```

Comment:
to prevent a huge nesting like this, you should reverse the conditional:
`if (!this.getInheritedPreferencesApi) { return null; }`